File: DS/Practice/linear_search.c
#include<stdio.h>
int main()
{
    int n, i, s, p = -1;

    //Taking input
    printf("Enter size of array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter array element:\n");
    for(i = 0; i < n; i++)
    {
        printf("arr[%d] = ", i);
        scanf("%d", &arr[i]);
    }
    printf("Enter element to search: ");
    scanf("%d", &s);
    for(i = 0; i < n; i++)
    {
        if(arr[i] == s)
        {
            p = i;
            printf("\nElement found at %dth index", p);
            break;
        }
    }
    if(p == -1)
    {
        printf("\n404: element not found");
    }
    return 0;
}

File: JAVA/Practise/LinearSearch.java
import java.util.Scanner;
class LinearSearch
{
	public static void main(String [] args)
	{
		Scanner sc = new Scanner(System.in);
		int n, s, pos = -1;
		System.out.print("Enter size of array: ");
		n = sc.nextInt();
		int arr[] = new int [n];
		for(int i = 0; i < n; i++)
		{
			System.out.print("arr[" + i + "] = ");
			arr[i] = sc.nextInt();
		}
		System.out.print("Enter searching element: ");
		s = sc.nextInt();
		for(int i = 0; i < n; i++)
		{
			if(arr[i] == s)
			{
				pos = i;
				break;
			}
		}
		if(pos == -1)
		{
			System.out.println("Element not found");
		}
		else
		{
			System.out.println("Element found at " + pos + " th index");
		}
	}
}